File: 프로그래머스/unrated/181833. 특별한 이차원 배열 1/특별한 이차원 배열 1.py
def solution(n):
    # 방법 2
    answer = [[0 for _ in range(n)] for _ in range(n)]
    
    for i in range(n):
        answer[i][i] = 1
    
    return answer

    # # 방법 1
    # return [[1 if i == j else 0 for j in range(n)] for i in range(n)]

File: 프로그래머스/unrated/181833. 특별한 이차원 배열 1/특별한 이차원 배열 1.js
function solution(n) {
    // 모든 원소가 0인 n행 n열 이중배열을 선언한다.
    const answer = Array.from(Array(n), () => Array(n).fill(0))
    
    // 행, 열의 인덱스가 같은 원소만 1로 바꾼다.
    for (let i = 0; i < n; i++) {
        answer[i][i] = 1
    }
    
    return answer
}